src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+  const [filterQuery, setFilterQuery] = useState('');
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const visibleTodos = todos.filter(todo => {
+    let isStatusCorrect = true;
+    const lowerCasedTitle = todo.title.toLowerCase();
+    const lowerCasedQuery = filterQuery.toLowerCase();
+
+    switch (filterStatus) {
+      case FilterStatus.Active:
+        isStatusCorrect = !todo.completed;
+        break;
+
+      case FilterStatus.Completed:
+        isStatusCorrect = todo.completed;
+        break;
+
+      default:
+        break;
+    }
+
+    return isStatusCorrect && lowerCasedTitle.includes(lowerCasedQuery);
+  });
```

Comment:
you can make this code better, by getting rid of this switch block, since it's not really necessary in your case, think on how to ;)